popdat/src/import_census.rs

```diff
@@ -1,128 +1,93 @@
-        //               }
-        //           },
-        //          ...
-        //      ]
-        // }
-        //
-        // Note: intuitively you might expect a collection of Polygon's rather than  MultiPolygons,
-        // but anecdotally, the census data I've seen uses MultiPolygons. In practice almost
-        // all are MultiPoly's with just one element, but some truly have multiple polygons.
-        //
-        // When we implement downloading, importer/src/utils.rs has a download() helper that we
-        // could copy here. (And later dedupe, after deciding how this crate will integrate with
-        // the importer)
-        let path = abstutil::path(format!(
-            "system/{}/population_areas.geojson",
-            map.get_name().city
-        ));
-        let bytes = abstutil::slurp_file(&path).map_err(|s| anyhow!(s))?;
-        debug!("parsing geojson at path: {}", &path);
-        timer.start("parsing geojson");
-        let geojson = GeoJson::from_reader(&*bytes)?;
-        timer.stop("parsing geojson");
-        let mut results = vec![];
-        let collection = geojson::FeatureCollection::try_from(geojson)?;
+    pub async fn fetch_all_for_map(
+        map_area: &Polygon,
+        bounds: &GPSBounds,
+    ) -> anyhow::Result<Vec<CensusArea>> {
+        use flatgeobuf::HttpFgbReader;
+        use geozero_core::geo_types::Geo;
 
-        let map_area = map.get_boundary_polygon();
-        let bounds = map.get_gps_bounds();
-
-        use geo::algorithm::map_coords::MapCoordsInplace;
+        use geo::algorithm::{bounding_rect::BoundingRect, map_coords::MapCoordsInplace};
         let mut geo_map_area: geo::Polygon<_> = map_area.clone().into();
         geo_map_area.map_coords_inplace(|c| {
             let projected = geom::Pt2D::new(c.0, c.1).to_gps(bounds);
             (projected.x(), projected.y())
         });
 
-        debug!("collection.features: {}", &collection.features.len());
-        timer.start("converting to `CensusArea`s");
-        for feature in collection.features.into_iter() {
-            let population = match feature
-                .property("population")
-                .ok_or(anyhow!("missing 'population' property"))?
-            {
-                serde_json::Value::Number(n) => n
-                    .as_u64()
-                    .ok_or(anyhow!("unexpected population number: {:?}", n))?
-                    as usize,
-                _ => {
-                    bail!(
-                        "unexpected format for 'population': {:?}",
-                        feature.property("population")
-                    );
+        // See the import handbook for how to prepare this file.
+        let mut fgb =
+            // HttpFgbReader::open("https://abstreet.s3.amazonaws.com/population_areas.fgb").await?;
+            HttpFgbReader::open("https://s3.amazonaws.com/mjk_asdf/abs/population_areas.fgb").await?;
```

Comment:
I changed this back to my bucket until your bucket has a cors policy setup.

Here's the one I used:
```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]
```

To set it up from the web: log in to the console > s3 > abstreet bucket > Permissions > Cors